### PR TITLE
Provide resource version in error if available

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
In store#Get, when key is not found, the composition of KeyNotFoundError uses hardcoded value of 0 for resource version.
This PR parses resourceVersion and uses the parsed value for KeyNotFoundError if the parsing is successful.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @wojtek-t

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 1]
```diff
@@ -124,6 +124,11 @@ func (s *store) Get(ctx context.Context, key string, resourceVersion string, out
 		if ignoreNotFound {
 			return runtime.SetZeroValue(out)
 		}
+		if len(resourceVersion) > 0 {
+			if rv, err := s.versioner.ParseResourceVersion(resourceVersion); err == nil {
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: resourceVersion was not used in the Get call to etcd and had no effect on the response-- I think it's confusing to return it like this. Can we revert this change?
